src/main/java/org/folio/fqm/repository/IdStreamer.java

```diff
@@ -91,19 +97,29 @@ public List<List<String>> getSortedIds(String derivedTableName,
   }
 
   private int streamIdsInBatch(EntityType entityType,
-                                boolean sortResults,
-                                Condition sqlWhereClause,
-                                int batchSize,
-                                Consumer<IdsWithCancelCallback> idsConsumer) {
+                               boolean sortResults,
+                               Condition sqlWhereClause,
+                               int batchSize,
+                               Consumer<IdsWithCancelCallback> idsConsumer) {
+    String finalJoinClause = entityTypeFlatteningService.getJoinClause(entityType);
     Field<String[]> idValueGetter = IdColumnUtils.getResultIdValueGetter(entityType);
+    String finalWhereClause = sqlWhereClause.toString();
+    for (EntityTypeSource source : entityType.getSources()) {
+      String toReplace = ":" + source.getAlias();
+      String alias = "\"" + source.getAlias() + "\"";
+      finalWhereClause = finalWhereClause.replace(toReplace, alias);
+    }
```

Comment:
We should do this replacement by using SQL parameters, if at all possible. Doing it manually is generally pretty dangerous and bug-prone. The code for handling `valueFunction` does it that way, so it could be a useful reference